api/src/opentrons/protocol_engine/state/liquids.py

```diff
@@ -0,0 +1,55 @@
+"""Basic liquid data state and store."""
+from dataclasses import dataclass
+from typing import Dict, List
+from opentrons.protocol_engine.types import Liquid
+
+from .abstract_store import HasState, HandlesActions
+from ..actions import Action, AddLiquidAction
+
+
+@dataclass
+class LiquidState:
+    """State of all loaded liquids."""
+
+    liquids_by_id: Dict[str, Liquid]
+
+
+class LiquidStore(HasState[LiquidState], HandlesActions):
+    """Liquid state container."""
+
+    _state: LiquidState
+
+    def __init__(self) -> None:
+        """Initialize a liquid store and its state."""
+        self._state = LiquidState(liquids_by_id={})
+
+    def handle_action(self, action: Action) -> None:
+        """Modify state in reaction to an action."""
+        if isinstance(action, AddLiquidAction):
+            self.add_liquid(action)
+
+    def add_liquid(self, action: AddLiquidAction) -> None:
```

Comment:
Make private or remove (a one-line method may not be worth the misdirection)

```suggestion
            self._add_liquid(action)

    def _add_liquid(self, action: AddLiquidAction) -> None:
```